macos build fails [cargo test]

Compiling on macos fails with a linker error.

```sh
sw_vers                                           

ProductName:		macOS
ProductVersion:		15.6
BuildVersion:		24G84
```

```
error: linking with `cc` failed: exit status: 1
  |
  = note:  "cc" "-Wl,-exported_symbols_list" "-Wl,/var/folders/_p/9pxdj7pd701c_dv2cx11g2ch0000gn/T/rustcr5X1Wu/list" "/var/folders/_p/9pxdj7pd701c_dv2cx11g2ch0000gn/T/rustcr5X1Wu/symbols.o" "<153 object files omitted>" "/Users/dan/proj/rust/bitemporal-timeseries/target/debug/deps/{libblake3-4ad05c2faea928d3.rlib,libconstant_time_eq-8cf44ff79543a08d.rlib,libarrayvec-584bcd4e9c207eff.rlib,libarrayref-822469aae5b80718.rlib,libordered_float-210c7c76501652d2.rlib,librayon-a5840b91327e088e.rlib,librayon_core-794ca03e77621cec.rlib,libcrossbeam_deque-11068498514484e8.rlib,libcrossbeam_epoch-d7a6f26ff82efd6c.rlib,libcrossbeam_utils-c9dfaf8c7be2fcfd.rlib,libeither-118adab2c119372e.rlib,libpyo3_arrow-19c2ef8f5490f4e9.rlib,libthiserror-e2bc16cdc865ddcb.rlib,libnumpy-37eb9a07d1cd7fef.rlib,librustc_hash-5881dbf718a3af75.rlib,libndarray-db873400ef22b0f4.rlib,libmatrixmultiply-8b481a034d8df014.rlib,librawpointer-5a2b441b717f9729.rlib,libpyo3-eec9d1b3b0dc4b95.rlib,libmemoffset-971a430334f79693.rlib,libportable_atomic-2ab890b5539f99e0.rlib,libparking_lot-749e6ca8bfa4eb7a.rlib,libparking_lot_core-caf334dd8a54ec2e.rlib,libsmallvec-a7cc77aa4515a555.rlib,liblock_api-a474742387a9a6f3.rlib,libscopeguard-3a94de5a59526d7e.rlib,libpyo3_ffi-e6a5201722a4fcaf.rlib,libunindent-af97e0e6dd8e8dab.rlib,libarrow-285bb1f2c0c061f5.rlib,libarrow_row-ae7473ad88a42583.rlib,libarrow_json-223d839de224bf21.rlib,libserde_json-0043c24668a8fd3f.rlib,libindexmap-254e60c7f36a1138.rlib,libequivalent-bd84ac2aecbfa2b1.rlib,libarrow_ipc-b05e460bb92153d9.rlib,libflatbuffers-69b6e1fa92c479ba.rlib,libbitflags-2037c6deb0b93138.rlib,libarrow_csv-7870065f2fe7fb9a.rlib,liblazy_static-3301e0e32549e5e3.rlib,libcsv-6c2cdc9454fe5770.rlib,libitoa-2ff3467e5794a240.rlib,libserde-33fb216e1b009487.rlib,libcsv_core-eaed2d9b368d57a6.rlib,libarrow_string-c7b03ef4b44f875a.rlib,libregex-22647c652ca76571.rlib,libregex_automata-956bf9fe3fe48b77.rlib,libaho_corasick-2b39588d6a30ad87.rlib,libregex_syntax-7fb552395858d7a0.rlib,libmemchr-5ec3f0e12fecfa8d.rlib,libarrow_cast-a43f95b007df783e.rlib,libatoi-fe807f42a3ae70bb.rlib,libryu-c8c07b8ff4b0babc.rlib,libbase64-af6d5c1450c31861.rlib,liblexical_core-9a6c7636c75a1698.rlib,liblexical_write_float-c4937adc74b86348.rlib,liblexical_write_integer-65529dcffa780ee5.rlib,liblexical_parse_float-cd3052c94c9919ec.rlib,liblexical_parse_integer-961c9a7fa93931d6.rlib,liblexical_util-b34fc2062a4f84c2.rlib,libstatic_assertions-c291e35e88be343e.rlib,libarrow_arith-d6b17e85c2f843d1.rlib,libarrow_ord-4d0357ab81365000.rlib,libarrow_select-172fca079b6ec0ef.rlib,libarrow_array-fd74a2cf5f5c13e9.rlib,libchrono_tz-aa7351f89be1e839.rlib,libphf-7b7d80b0caa5b2c9.rlib,libphf_shared-e24e38b4f137c065.rlib,libsiphasher-514260552170d939.rlib,libahash-c639c1537090512d.rlib,libgetrandom-7083af508278d35d.rlib,liblibc-a1418ffcb821d730.rlib,libonce_cell-5d51a55a1a353586.rlib,libzerocopy-eeecf2d00df16fd3.rlib,libhashbrown-58e51d2a84578130.rlib,libchrono-e49abb7a62d22421.rlib,libiana_time_zone-f8009028437d5606.rlib,libcore_foundation_sys-f232f6c2ad27bdc8.rlib,libarrow_data-436bf159f7ae7dcb.rlib,libarrow_schema-69ce01f7834b18af.rlib,libbitflags-31f633feba577481.rlib,libarrow_buffer-4cce2f2e4985c10e.rlib,libbytes-8029efeda244a01c.rlib,libhalf-fd2484b2b920e7bb.rlib,libcfg_if-69f417a988bf63b5.rlib,libnum-a4a4daf22ef6c226.rlib,libnum_iter-68ecbf927470ada9.rlib,libnum_rational-08d5a14fffd67a60.rlib,libnum_complex-b8c61e0744b09487.rlib,libnum_bigint-dd7b4c6fb2e7773c.rlib,libnum_integer-cede6b71dcc34291.rlib,libnum_traits-607959a14b3d3a8a.rlib}.rlib" "<sysroot>/lib/rustlib/aarch64-apple-darwin/lib/{libstd-*,libpanic_unwind-*,libobject-*,libmemchr-*,libaddr2line-*,libgimli-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libminiz_oxide-*,libadler2-*,libunwind-*,libcfg_if-*,liblibc-*,librustc_std_workspace_core-*,liballoc-*,libcore-*,libcompiler_builtins-*}.rlib" "-liconv" "-framework" "CoreFoundation" "-lSystem" "-lc" "-lm" "-arch" "arm64" "-mmacosx-version-min=11.0.0" "-L" "/Users/dan/proj/rust/bitemporal-timeseries/target/debug/build/blake3-40380c01957e59de/out" "-o" "/Users/dan/proj/rust/bitemporal-timeseries/target/debug/deps/libbitemporal_timeseries.dylib" "-Wl,-dead_strip" "-dynamiclib" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: Undefined symbols for architecture arm64:
            "_PyBaseObject_Type", referenced from:
                _$LT$pyo3..types..any..PyAny$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h4ce077881e6cf892 in libnumpy-37eb9a07d1cd7fef.rlib[5](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.2.rcgu.o)
                _$LT$pyo3..pycell..impl_..PyClassObjectBase$LT$U$GT$$u20$as$u20$pyo3..pycell..impl_..PyClassObjectLayout$LT$T$GT$$GT$::tp_dealloc::he3dd0a009430cece in libnumpy-37eb9a07d1cd7fef.rlib[6](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.3.rcgu.o)
                _$LT$pyo3..pyclass_init..PyNativeTypeInitializer$LT$T$GT$$u20$as$u20$pyo3..pyclass_init..PyObjectInit$LT$T$GT$$GT$::into_new_object::inner::h46a1a27ca801dbc4 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "_PyBool_Type", referenced from:
                pyo3_ffi::object::Py_SIZE::h195b1a478b465a06 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "_PyBytes_AsString", referenced from:
                pyo3::types::bytes::_$LT$impl$u20$pyo3..instance..Borrowed$LT$pyo3..types..bytes..PyBytes$GT$$GT$::as_bytes::hf779f15bd90049c9 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyBytes_Size", referenced from:
                pyo3::types::bytes::_$LT$impl$u20$pyo3..instance..Borrowed$LT$pyo3..types..bytes..PyBytes$GT$$GT$::as_bytes::hf779f15bd90049c9 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyCMethod_New", referenced from:
                pyo3_ffi::methodobject::PyCFunction_NewEx::hed0862d7f0e643db in libpyo3-eec9d1b3b0dc4b95.rlib[11](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.08.rcgu.o)
            "_PyCapsule_GetContext", referenced from:
                pyo3::types::capsule::capsule_destructor::h5ac02dffd23c7090 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                pyo3::types::capsule::capsule_destructor::hf43f76db7a670c6c in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
            "_PyCapsule_GetName", referenced from:
                pyo3::types::capsule::capsule_destructor::h5ac02dffd23c7090 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                pyo3::types::capsule::capsule_destructor::hf43f76db7a670c6c in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$pyo3..types..capsule..PyCapsule$GT$$u20$as$u20$pyo3..types..capsule..PyCapsuleMethods$GT$::name::h5db57d23793faf78 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                pyo3::types::capsule::name_ptr_ignore_error::h4f6c90b5b0420af5 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyCapsule_GetPointer", referenced from:
                pyo3::types::capsule::capsule_destructor::h5ac02dffd23c7090 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                pyo3::types::capsule::capsule_destructor::hf43f76db7a670c6c in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$pyo3..types..capsule..PyCapsule$GT$$u20$as$u20$pyo3..types..capsule..PyCapsuleMethods$GT$::pointer::h9dad3e49d5269746 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyCapsule_New", referenced from:
                pyo3::types::capsule::PyCapsule::new_bound_with_destructor::h9c234bfa400b404d in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                pyo3::types::capsule::PyCapsule::new_bound_with_destructor::hdafd350bda5a8321 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
            "_PyCapsule_Type", referenced from:
                pyo3_ffi::pycapsule::PyCapsule_CheckExact::h2066cca45c39cafb in libnumpy-37eb9a07d1cd7fef.rlib[7](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.4.rcgu.o)
            "_PyDict_Next", referenced from:
                _$LT$pyo3..types..dict..borrowed_iter..BorrowedDictIter$u20$as$u20$core..iter..traits..iterator..Iterator$GT$::next::hd186f854489b85b4 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                _$LT$pyo3..types..dict..BoundDictIterator$u20$as$u20$core..iter..traits..iterator..Iterator$GT$::next::h11e508493304b807 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
            "_PyErr_Clear", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..capsule..PyCapsule$GT$$u20$as$u20$pyo3..types..capsule..PyCapsuleMethods$GT$::pointer::h9dad3e49d5269746 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                pyo3::types::capsule::name_ptr_ignore_error::h4f6c90b5b0420af5 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyErr_GetRaisedException", referenced from:
                pyo3::err::err_state::PyErrStateNormalized::take::h0da64c41f41f931a in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyErr_GivenExceptionMatches", referenced from:
                pyo3::err::PyErr::is_instance_bound::h0752ad655869e50c in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyErr_NewExceptionWithDoc", referenced from:
                pyo3::err::PyErr::new_type_bound::h537bf614581b7fb7 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyErr_Print", referenced from:
                pyo3::err::panic_after_error::h7ebad6f44db57756 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyErr_PrintEx", referenced from:
                pyo3::err::PyErr::print_panic_and_unwind::h1f25d5608edaa6d5 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
                pyo3::err::PyErr::print::had712b2d55202550 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyErr_SetObject", referenced from:
                pyo3::err::err_state::raise_lazy::h874506e116672c84 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyErr_SetRaisedException", referenced from:
                pyo3::err::err_state::PyErrState::restore::h6f959e1f1b9d954c in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyErr_SetString", referenced from:
                pyo3::err::err_state::raise_lazy::h874506e116672c84 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyErr_WriteUnraisable", referenced from:
                pyo3::err::PyErr::write_unraisable_bound::hea5b8960ba5817d7 in libnumpy-37eb9a07d1cd7fef.rlib[6](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.3.rcgu.o)
                pyo3::err::PyErr::write_unraisable_bound::hbdd81ba4bd91ce74 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyEval_SaveThread", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::head1d4600043aa4d in libpyo3-eec9d1b3b0dc4b95.rlib[7](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.04.rcgu.o)
            "_PyExc_AttributeError", referenced from:
                _$LT$pyo3..exceptions..PyAttributeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h6e23b172ca25f12d in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyAttributeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h6e23b172ca25f12d in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_BaseException", referenced from:
                _$LT$pyo3..exceptions..PyBaseException$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::he61cfe28d82e73bd in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyBaseException$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::he61cfe28d82e73bd in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_Exception", referenced from:
                _$LT$pyo3..exceptions..PyException$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h2b48ed2c1125a927 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[16](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.13.rcgu.o)
                _$LT$pyo3..exceptions..PyException$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h2b48ed2c1125a927 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[16](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.13.rcgu.o)
            "_PyExc_IOError", referenced from:
                _$LT$pyo3..exceptions..PyIOError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h5a50e4525e9a38b7 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[16](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.13.rcgu.o)
                _$LT$pyo3..exceptions..PyIOError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h5a50e4525e9a38b7 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[16](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.13.rcgu.o)
            "_PyExc_ImportError", referenced from:
                _$LT$pyo3..exceptions..PyImportError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h77c57c01e3427e9d in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyImportError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h77c57c01e3427e9d in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_OverflowError", referenced from:
                _$LT$pyo3..exceptions..PyOverflowError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h610ee73e03cff426 in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyOverflowError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h610ee73e03cff426 in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_RuntimeError", referenced from:
                _$LT$pyo3..exceptions..PyRuntimeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::hbebbc06e7f02e677 in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyRuntimeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::hbebbc06e7f02e677 in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_TypeError", referenced from:
                _$LT$pyo3..exceptions..PyTypeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h5133c5fb9ccba3ce in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyTypeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h5133c5fb9ccba3ce in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_UnicodeDecodeError", referenced from:
                _$LT$pyo3..exceptions..PyUnicodeDecodeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::hc283bac4104b337c in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyUnicodeDecodeError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::hc283bac4104b337c in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyExc_ValueError", referenced from:
                _$LT$pyo3..exceptions..PyValueError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h88b1fe32af4be016 in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
                _$LT$pyo3..exceptions..PyValueError$u20$as$u20$pyo3..type_object..PyTypeInfo$GT$::type_object_raw::_$u7b$$u7b$closure$u7d$$u7d$::h88b1fe32af4be016 in libpyo3-eec9d1b3b0dc4b95.rlib[4](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.01.rcgu.o)
            "_PyException_GetCause", referenced from:
                pyo3::err::PyErr::cause::he15f8268b71010bf in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyException_GetTraceback", referenced from:
                pyo3::err::err_state::PyErrStateNormalized::ptraceback::hf1d06a386130e733 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyException_SetCause", referenced from:
                pyo3::err::PyErr::set_cause::h89caa1815bfd700a in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyException_SetTraceback", referenced from:
                pyo3::err::PyErr::into_value::h9d6843b0ba7253d5 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyGILState_Ensure", referenced from:
                pyo3::gil::GILGuard::acquire_unchecked::h884d0ebbfe8ee30b in libpyo3-eec9d1b3b0dc4b95.rlib[7](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.04.rcgu.o)
            "_PyGILState_Release", referenced from:
                _$LT$pyo3..gil..GILGuard$u20$as$u20$core..ops..drop..Drop$GT$::drop::h9c1546baedeee560 in libpyo3-eec9d1b3b0dc4b95.rlib[7](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.04.rcgu.o)
            "_PyImport_Import", referenced from:
                pyo3::types::module::PyModule::import_bound::h6b3f170bf2699bd3 in libpyo3-eec9d1b3b0dc4b95.rlib[6](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.03.rcgu.o)
            "_PyInterpreterState_Get", referenced from:
                pyo3::impl_::pymodule::ModuleDef::make_module::hb847091d37d59e7f in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyInterpreterState_GetID", referenced from:
                pyo3::impl_::pymodule::ModuleDef::make_module::hb847091d37d59e7f in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyIter_Next", referenced from:
                pyo3::types::iterator::_$LT$impl$u20$pyo3..instance..Borrowed$LT$pyo3..types..iterator..PyIterator$GT$$GT$::next::h680d2f119bec2a03 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyList_Append", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..list..PyList$GT$$u20$as$u20$pyo3..types..list..PyListMethods$GT$::append::inner::h5077b2006ed24129 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "_PyList_New", referenced from:
                pyo3::types::list::new_from_iter::h8134e3000863b5d3 in bitemporal_timeseries.3ffkv0skzjuhodaxvb8sos98b.0mnmnb4.rcgu.o
                pyo3::types::list::new_from_iter::h69e394997baa6ee5 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[6](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.03.rcgu.o)
                pyo3::types::list::PyList::empty_bound::h34b8505dd2fdbf00 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "_PyLong_AsLong", referenced from:
                pyo3::conversions::std::num::_$LT$impl$u20$pyo3..conversion..FromPyObject$u20$for$u20$u8$GT$::extract_bound::h22ff3ad4657e473e in libpyo3-eec9d1b3b0dc4b95.rlib[15](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.12.rcgu.o)
            "_PyLong_AsUnsignedLongLong", referenced from:
                pyo3::conversions::std::num::_$LT$impl$u20$pyo3..conversion..FromPyObject$u20$for$u20$u64$GT$::extract_bound::hb5a76fab15968baa in libpyo3-eec9d1b3b0dc4b95.rlib[15](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.12.rcgu.o)
                pyo3::conversions::std::num::_$LT$impl$u20$pyo3..conversion..FromPyObject$u20$for$u20$u64$GT$::extract_bound::hb5a76fab15968baa in libpyo3-eec9d1b3b0dc4b95.rlib[15](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.12.rcgu.o)
            "_PyLong_FromLong", referenced from:
                pyo3::conversions::std::num::_$LT$impl$u20$pyo3..conversion..ToPyObject$u20$for$u20$i32$GT$::to_object::h61b444bfd00fac78 in libpyo3-eec9d1b3b0dc4b95.rlib[15](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.12.rcgu.o)
            "_PyLong_FromSsize_t", referenced from:
                pyo3::impl_::pyclass::get_sequence_item_from_mapping::h2a286ffd72363749 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
                pyo3::impl_::pyclass::assign_sequence_item_from_mapping::ha040d28d1688be3f in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyLong_FromUnsignedLongLong", referenced from:
                pyo3::conversions::std::num::_$LT$impl$u20$pyo3..conversion..IntoPy$LT$pyo3..instance..Py$LT$pyo3..types..any..PyAny$GT$$GT$$u20$for$u20$u64$GT$::into_py::h06f67141f1a021c5 in libpyo3-eec9d1b3b0dc4b95.rlib[15](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.12.rcgu.o)
            "_PyLong_Type", referenced from:
                pyo3_ffi::object::Py_SIZE::h195b1a478b465a06 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "_PyModule_Create2", referenced from:
                pyo3_ffi::modsupport::PyModule_Create::h7a14fa96006c250d in libpyo3-eec9d1b3b0dc4b95.rlib[6](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.03.rcgu.o)
            "_PyModule_GetNameObject", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..module..PyModule$GT$$u20$as$u20$pyo3..types..module..PyModuleMethods$GT$::name::h3a8118c89b252061 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyNumber_Index", referenced from:
                pyo3::conversions::std::num::_$LT$impl$u20$pyo3..conversion..FromPyObject$u20$for$u20$u64$GT$::extract_bound::hb5a76fab15968baa in libpyo3-eec9d1b3b0dc4b95.rlib[15](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.12.rcgu.o)
            "_PyObject_Call", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::call::inner::h6507e4cf9399b4c9 in libpyo3-eec9d1b3b0dc4b95.rlib[12](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.09.rcgu.o)
            "_PyObject_CallNoArgs", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::call0::hbf806ce8e2dfcd9a in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyObject_DelItem", referenced from:
                pyo3::impl_::pyclass::assign_sequence_item_from_mapping::ha040d28d1688be3f in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyObject_GC_UnTrack", referenced from:
                pyo3::impl_::pyclass::tp_dealloc_with_gc::h29447bdcb6621d0d in libpyo3_arrow-19c2ef8f5490f4e9.rlib[11](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.08.rcgu.o)
            "_PyObject_GenericGetDict", referenced from:
                pyo3::pyclass::create_type_object::PyTypeBuilder::finalize_methods_and_properties::h7c25fcee70a99654 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyObject_GenericSetDict", referenced from:
                pyo3::pyclass::create_type_object::PyTypeBuilder::finalize_methods_and_properties::h7c25fcee70a99654 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyObject_GetAttr", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::getattr::inner::h422d3c58ad5fc889 in libpyo3-eec9d1b3b0dc4b95.rlib[12](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.09.rcgu.o)
            "_PyObject_GetItem", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::get_item::inner::h0d3440ab6269f7f5 in libpyo3-eec9d1b3b0dc4b95.rlib[12](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.09.rcgu.o)
                pyo3::impl_::pyclass::get_sequence_item_from_mapping::h2a286ffd72363749 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyObject_GetIter", referenced from:
                pyo3::types::iterator::PyIterator::from_bound_object::hbea4f6c0c57ad40c in libpyo3-eec9d1b3b0dc4b95.rlib[14](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.11.rcgu.o)
            "_PyObject_IsTrue", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::is_truthy::h2de45e6950e971f3 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyObject_Repr", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::repr::h0a62968e4794b40b in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyObject_RichCompare", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::rich_compare::inner::h11805051510fa850 in libpyo3-eec9d1b3b0dc4b95.rlib[12](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.09.rcgu.o)
            "_PyObject_SetAttr", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::setattr::inner::h8b783e9dcd2daf67 in libpyo3-eec9d1b3b0dc4b95.rlib[12](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.09.rcgu.o)
            "_PyObject_SetAttrString", referenced from:
                pyo3::impl_::pyclass::lazy_type_object::initialize_tp_dict::h490ada8ac2656217 in libpyo3-eec9d1b3b0dc4b95.rlib[8](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.05.rcgu.o)
            "_PyObject_SetItem", referenced from:
                pyo3::impl_::pyclass::assign_sequence_item_from_mapping::ha040d28d1688be3f in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyObject_Str", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..any..PyAny$GT$$u20$as$u20$pyo3..types..any..PyAnyMethods$GT$::str::h1659f0fc4983eb6a in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PySequence_Check", referenced from:
                pyo3::types::sequence::extract_sequence::h77e7b1753d104e35 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[5](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.02.rcgu.o)
                pyo3::types::sequence::extract_sequence::hcbe51b79a2942b8e in libpyo3_arrow-19c2ef8f5490f4e9.rlib[5](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.02.rcgu.o)
                pyo3::types::sequence::extract_sequence::ha5c8a08e3e5f250f in libnumpy-37eb9a07d1cd7fef.rlib[6](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.3.rcgu.o)
                pyo3::types::sequence::extract_sequence::hddd035f77c80c3ec in libnumpy-37eb9a07d1cd7fef.rlib[6](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.3.rcgu.o)
            "_PySequence_Size", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..sequence..PySequence$GT$$u20$as$u20$pyo3..types..sequence..PySequenceMethods$GT$::len::h8fb83d0b2f1d6cc6 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$pyo3..types..sequence..PySequence$GT$$u20$as$u20$pyo3..types..sequence..PySequenceMethods$GT$::len::hfa97f16fe52f2cc2 in libnumpy-37eb9a07d1cd7fef.rlib[8](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.5.rcgu.o)
            "_PyTuple_New", referenced from:
                pyo3::types::tuple::new_from_iter::ha6bf27d066b1aa78 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[8](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.05.rcgu.o)
                pyo3::types::tuple::array_into_tuple::h21458f49287b22a1 in libpyo3-eec9d1b3b0dc4b95.rlib[14](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.11.rcgu.o)
                pyo3::types::tuple::array_into_tuple::hd7edb1ae9991ab0c in libpyo3-eec9d1b3b0dc4b95.rlib[14](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.11.rcgu.o)
            "_PyType_FromSpec", referenced from:
                pyo3::pyclass::create_type_object::PyTypeBuilder::build::h4e6d36cfeab59c9f in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyType_GenericAlloc", referenced from:
                _$LT$pyo3..pyclass_init..PyNativeTypeInitializer$LT$T$GT$$u20$as$u20$pyo3..pyclass_init..PyObjectInit$LT$T$GT$$GT$::into_new_object::inner::h46a1a27ca801dbc4 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "_PyType_GetQualName", referenced from:
                _$LT$pyo3..instance..Bound$LT$pyo3..types..typeobject..PyType$GT$$u20$as$u20$pyo3..types..typeobject..PyTypeMethods$GT$::qualname::h09272048c62ff20e in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyType_IsSubtype", referenced from:
                pyo3_ffi::object::PyObject_TypeCheck::hddace49cbc8f7eb9 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[6](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.03.rcgu.o)
            "_PyUnicode_AsEncodedString", referenced from:
                pyo3::types::string::_$LT$impl$u20$pyo3..instance..Borrowed$LT$pyo3..types..string..PyString$GT$$GT$::to_string_lossy::hb6059c33aff35046 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyUnicode_AsUTF8AndSize", referenced from:
                pyo3::types::string::_$LT$impl$u20$pyo3..instance..Borrowed$LT$pyo3..types..string..PyString$GT$$GT$::to_str::h7e76ff3669df5e58 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
            "_PyUnicode_FromStringAndSize", referenced from:
                pyo3::types::string::PyString::new_bound::h47a40e17d5301154 in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
                pyo3::types::string::PyString::intern_bound::h03e791350120fc4e in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_PyUnicode_InternInPlace", referenced from:
                pyo3::types::string::PyString::intern_bound::h03e791350120fc4e in libpyo3-eec9d1b3b0dc4b95.rlib[17](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.14.rcgu.o)
            "_Py_InitializeEx", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::head1d4600043aa4d in libpyo3-eec9d1b3b0dc4b95.rlib[7](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.04.rcgu.o)
            "_Py_IsInitialized", referenced from:
                pyo3::gil::prepare_freethreaded_python::_$u7b$$u7b$closure$u7d$$u7d$::head1d4600043aa4d in libpyo3-eec9d1b3b0dc4b95.rlib[7](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.04.rcgu.o)
                pyo3::gil::GILGuard::acquire::_$u7b$$u7b$closure$u7d$$u7d$::h30e72c33b55997ce in libpyo3-eec9d1b3b0dc4b95.rlib[7](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.04.rcgu.o)
            "__Py_Dealloc", referenced from:
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h471c5e463b700256 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[15](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.12.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h385cfecf758810e9 in libnumpy-37eb9a07d1cd7fef.rlib[8](numpy-37eb9a07d1cd7fef.numpy.1bcbb0ae1f0f7f36-cgu.5.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h0089395b423aff77 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h14c4d3176ec324e0 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h1eeea831f1cb3f05 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h5ae3168f9e0a4481 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                _$LT$pyo3..instance..Bound$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h60cbb4c51869a5c2 in libpyo3-eec9d1b3b0dc4b95.rlib[3](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.00.rcgu.o)
                ...
            "__Py_FalseStruct", referenced from:
                pyo3_ffi::boolobject::Py_False::haea3f0e544dc8015 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[17](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.14.rcgu.o)
            "__Py_NoneStruct", referenced from:
                pyo3_ffi::object::Py_None::h9c25536980e2e9fc in libpyo3_arrow-19c2ef8f5490f4e9.rlib[6](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.03.rcgu.o)
                pyo3_ffi::object::Py_None::ha2ca3058f245a529 in libpyo3-eec9d1b3b0dc4b95.rlib[16](pyo3-eec9d1b3b0dc4b95.pyo3.cbe705faab1b37ea-cgu.13.rcgu.o)
            "__Py_NotImplementedStruct", referenced from:
                pyo3_ffi::object::Py_NotImplemented::h640b7a1ac9a64595 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[6](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.03.rcgu.o)
            "__Py_TrueStruct", referenced from:
                pyo3_ffi::boolobject::Py_True::h6b4213b9018f7b75 in libpyo3_arrow-19c2ef8f5490f4e9.rlib[17](pyo3_arrow-19c2ef8f5490f4e9.pyo3_arrow.ddeeb46a65482232-cgu.14.rcgu.o)
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `bitemporal-timeseries` (lib) due to 1 previous error
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -5,7 +5,7 @@ edition = "2021"
 
 [dependencies]
 arrow = "53.4"
-pyo3 = { version = "0.21", features = ["extension-module"] }
+pyo3 = "0.21"
 pyo3-arrow = "0.3"
 chrono = "0.4"
 blake3 = "1.5"
@@ -15,6 +15,9 @@ ordered-float = "4.2"
 [dev-dependencies]
 criterion = { version = "0.5", features = ["html_reports", "cargo_bench_support"] }
 pprof = { version = "0.13", features = ["flamegraph", "criterion"] }
+
+[features]
+py-ext = ["pyo3/extension-module"]
 
 [lib]
 name = "bitemporal_timeseries"
@@ -26,3 +29,4 @@ lto = true
 [[bench]]
 name = "bitemporal_benchmarks"
 harness = false
+

